include/seastar/rpc/rpc_types.hh

```diff
@@ -290,7 +290,7 @@ public:
         // return feature string that will be sent as part of protocol negotiation
         virtual const sstring& supported() const = 0;
         // negotiate compress algorithm
-        virtual std::unique_ptr<compressor> negotiate(sstring feature, bool is_server, condition_variable* compressor_needs_progress) {
+        virtual std::unique_ptr<compressor> negotiate(sstring feature, bool is_server, condition_variable* compressor_needs_progress) const {
```

Comment:
Why not squash with the previous patch?

Reply:
Why do? Usually we like to split patches into smaller ones when possible.

Reply:
Because you introduced this function in the previous patch and add const to it now. Why not add it from the beginning?

Reply:
Ah, you meant only the `const`.

Right, this is a leftover from an earlier revision, I'll fold the `const` into the previous patch.